### PR TITLE
Created error HOC to display html errors on the site when they occure

### PR DESCRIPTION
**Pull request by BastiaanReinalda**
Implemented the HOC into Log.tsx, Logs.tsx, Run.tsx and Runs.tsx.

### CODE REVIEW

File: src/app/models/Log.tsx [hunk 1]
```diff
@@ -21,98 +21,53 @@ export interface Log {
...
-            method: 'POST',
-            url: `${API_URL}logs`,
-            data: LogModel.createLog,
-            withCredentials: false
-        });
+    createLog: {
+        error: '',
+        log: {} as Log,
+        async save() {
+            LogModel.createLog.log.creationTime = new Date().toString();
```
Comment (RamonG92) [inline src/app/models/Log.tsx]: Shouldn't the 'creationTime' be done in the api instead in the front-end.

File: src/app/models/Log.tsx [hunk 2]
```diff
@@ -21,98 +21,53 @@ export interface Log {
...
+            LogModel.createLog.log.origin = 'human';
```
Comment (RamonG92) [inline src/app/models/Log.tsx]: Should it be the same this?

File: src/app/models/Log.tsx [hunk 3]
```diff
@@ -21,98 +21,53 @@ export interface Log {
...
-            method: 'GET',
-            url: `${API_URL}logs/${id}`,
-            withCredentials: false
-        }).then((result: any) => {
-            this.current = result;
-        }).catch(() => {
-            const result = dummyLogList[0];
-            this.current = result;
-        });
+    current: {
```
Comment (RamonG92) [inline src/app/models/Log.tsx]: Maybe this should have a different name, one that's more obvious?

File: src/app/models/Run.tsx [hunk 4]
```diff
@@ -28,35 +28,49 @@ export interface Run {
...
     },
-    fetchById(id: number) {
-        return m.request<Run>({
-            method: 'GET',
-            url: `${API_URL}runs/${id}`,
-            withCredentials: false
-        }).then((result: any) => {
-            this.current = result;
-        });
+    current: {
```
Comment (RamonG92) [inline src/app/models/Run.tsx]: Maybe this should have a different name, one that's more obvious?

File: src/app/models/Run.tsx [hunk 5]
```diff
@@ -28,35 +28,49 @@ export interface Run {
...
-    async fetch(query?: string) {
-        return m.request({
-            method: 'GET',
-            url: `${API_URL}runs${query ? `?${query}` : ''}`,
-            withCredentials: false
-        }).then((result: any) => {
-            console.log('fetching in run');
-            this.list = result;
-        });
+    list: {
```
Comment (RamonG92) [inline src/app/models/Run.tsx]: Maybe this should have a different name, one that's more obvious?

File: src/app/models/Log.tsx [hunk 6]
```diff
@@ -21,98 +21,53 @@ export interface Log {
...
-            method: 'GET',
-            url: `${API_URL}logs${query ? `?${query}` : ''}`,
-            withCredentials: false
-        }).then((result: any) => {
-            this.list = result;
-        }).catch(() => {
-            const result = dummyLogList;
-            this.list = result;
-        });
+    list: {
```
Comment (RamonG92) [inline src/app/models/Log.tsx]: Maybe this should have a different name, one that's more obvious?

File: src/app/models/Log.tsx (hunk 1 above)
Comment (BastiaanReinalda) [reply src/app/models/Log.tsx]: I am not really sure what we decided to do, but i think it should, yes. But it is not yet implemented in the api yet.

File: src/app/models/Log.tsx (hunk 2 above)
Comment (BastiaanReinalda) [reply src/app/models/Log.tsx]: I am not sure what you mean here?

File: src/app/models/Log.tsx (hunk 6 above)
Comment (BastiaanReinalda) [reply src/app/models/Log.tsx]: What should have a different name?

File: src/app/models/Log.tsx (hunk 2 above)
Comment (RamonG92) [reply src/app/models/Log.tsx]: The same as with creationTime, that it should be in the api instead of fe?

File: src/app/models/Log.tsx (hunk 1 above)
Comment (misharigot) [reply src/app/models/Log.tsx]: Out of the card's scope, Im working on removing front end date creation right now. 
image

**Pull request merged**